### ISSUE TITLE
node has data and has sub nodes question

### ISSUE THREAD

**Issue by killmenot** [robtweed/ewd-document-store#11]
@robtweed Please take a look:
https://github.com/robtweed/ewd-document-store/blob/a771f87d7d6e09bf28719b3be8cd139054dd9c0f/lib/proto/getDocument.js#L96-L104

What if node has data and has sub nodes (defined = 11)? In this case data set on line 97 was replaced on line 103. Is it correct?

**Comment by robtweed** [comment]
Yes - Global storage differs from JSON storage in that a Global can have a data value at an intermediate node.  JSON can only have data at a leaf node.  My decision was to ignore any data that might exist in an intermediate node and continue down the node hierarchy, when converting to a JSON document

**Comment by killmenot** [comment]
@robtweed I may be wrong but  seems we can skip call
```js
document[index] = db.get(node).data
```
for `defined = 11` because it will be overwritten anyway, isn't it? We need `db.get` only for `defined = 1`. Please correct me if I'm wrong

**Comment by robtweed** [comment]
I suspect you're right

**Comment by robtweed** [comment]
I'll fix in next build update
